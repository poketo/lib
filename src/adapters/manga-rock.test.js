import poketo from '..';
import adapter from './manga-rock';

describe('MangaRockAdapter', () => {
  describe('parseUrl', () => {
    it('returns the components of a url', () => {
      expect(
        adapter.parseUrl('https://mangarock.com/manga/mrs-serie-100056453'),
      ).toEqual({
        seriesSlug: '100056453',
        chapterSlug: null,
      });

      expect(
        adapter.parseUrl(
          'https://mangarock.com/manga/mrs-serie-100056453/chapter/mrs-chapter-100259395',
        ),
      ).toEqual({
        seriesSlug: '100056453',
        chapterSlug: '100259395',
      });
    });

    it('throws on unparseable urls', () => {
      expect(() => {
        adapter.parseUrl('https://mangarock.com/author');
      }).toThrow(poketo.InvalidUrlError);
    });
  });

  describe('constructUrl', () => {
    it('returns a valid url', () => {
      expect(adapter.constructUrl('295440')).toEqual(
        'https://mangarock.com/manga/mrs-serie-295440',
      );

      expect(adapter.constructUrl('295440', '100197426')).toEqual(
        'https://mangarock.com/manga/mrs-serie-295440/chapter/mrs-chapter-100197426',
      );
    });
  });
});

describe('MangaRock', () => {
  const server = new AdapterVcrServer(adapter);

  beforeAll(async () => {
    await server.listenAndMock();
  });

  afterAll(() => {
    server.close();
  });

  describe('getSeries', async () => {
    it('returns a metadata object', async () => {
      const { chapters, ...metadata } = await poketo.getSeries(
        'manga-rock:153037',
      );

      expect(metadata).toMatchSnapshot({
        updatedAt: expect.any(Number),
      });
    });
  });

  describe('getChapter', () => {
    it('returns a list of pages', async () => {
      const chapter = await poketo.getChapter('manga-rock:100056453:100259395');

      expect(chapter.pages).toMatchSnapshot();
    });
  });
});
