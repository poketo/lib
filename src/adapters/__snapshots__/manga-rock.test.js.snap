// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MangaRock getChapter returns a list of pages 1`] = `
Array [
  Object {
    "id": "T3.3BZTN7p5.mri",
    "url": "https://mri-image-decoder.now.sh/?url=https://f01.mrcdn.info/file/mrfiles/i/6/n/l/T3.3BZTN7p5.mri",
  },
]
`;

exports[`MangaRock getSeries returns a metadata object 1`] = `
Object {
  "coverImageUrl": "https://f01.mrcdn.info/file/mrportal/h/9/4/m/cC.btfAGMrS.jpg",
  "id": "manga-rock:153037",
  "site": Object {
    "id": "manga-rock",
    "name": "Manga Rock",
  },
  "slug": "153037",
  "supportsReading": true,
  "title": "Love So Life",
  "updatedAt": Any<Number>,
  "url": "https://mangarock.com/manga/mrs-serie-153037",
}
`;
